src/test/core/CryptoPRNG_test.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx/Env.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/crypto/csprng.h>
+#include <boost/scope_exit.hpp>
+#include <boost/filesystem.hpp>
+#include <fstream>
+
+namespace ripple {
+
+class CryptoPRNG_test : public beast::unit_test::suite
+{
+    void
+    testGetValues()
+    {
+        testcase ("Get Values");
+        try
+        {
+            auto& engine = crypto_prng();
+            auto rand_val = engine();
+            BEAST_EXPECT(rand_val >= engine.min());
+            BEAST_EXPECT(rand_val <= engine.max());
+
+            uint16_t twoByte {0};
+            engine(&twoByte, sizeof(uint16_t));
+            pass();
+        }
+        catch(std::exception&)
+        {
+            fail();
+        }
+    }
+
+    class scopedTempDir
+    {
+        boost::filesystem::path path_;
+        beast::unit_test::suite& test_;
+
+    public:
+        scopedTempDir (beast::unit_test::suite& suite) : test_(suite)
+        {
+            using namespace boost::filesystem;
+            // use the throwing version of these functions
+            // so ctor will bail on error
+            path_  = temp_directory_path();
+            path_ /= unique_path("%%%%-%%%%-%%%%-%%%%");
+            create_directories(path_);
+        };
+
+        ~scopedTempDir()
+        {
+            // use non-throwing calls here
+            boost::system::error_code ec;
+            boost::filesystem::remove_all(path_, ec);
+            if(ec)
+                test_.log << "WARNING - temp dir not removed: " << ec.message();
+        };
+
+        auto const& getPath() { return path_; };
+    };
+
+    void
+    testSaveLoad()
+    {
+        testcase ("Save/Load State");
+        try
+        {
+            // create a temporary path to write crypto state files
+            scopedTempDir td {*this};
+
+            auto stateFile = td.getPath() / "cryptostate";
+            auto& engine = crypto_prng();
+            engine.mix_entropy();
```

Comment:
https://github.com/ripple/rippled/blob/develop/src/ripple/crypto/impl/csprng.cpp#L75 is the implementation - but I think you might also be pointing out that the engine constructor already calls `mix_entropy()` so what's the point of calling it again(?) I agree, not much point in calling it explicitly - I can remove this.